tony-azkaban/src/main/java/com/linkedin/tony/azkaban/TensorFlowJob.java

```diff
@@ -0,0 +1,195 @@
+  private boolean shouldProxy = false;
+  private boolean obtainTokens = false;
+  private File tokenFile = null;
+  public static final String HADOOP_OPTS = ENV_PREFIX + "HADOOP_OPTS";
+  public static final String HADOOP_GLOBAL_OPTS = "hadoop.global.opts";
+  public static final String WORKER_ENV_PREFIX = "worker_env.";
+  private static final String TONY_XML = "_tony-conf/tony.xml";
+  private static final String TONY_CONF_PREFIX = "tony.";
+  private HadoopSecurityManager hadoopSecurityManager;
+  private File tonyConfFile;
+
+  public TensorFlowJob(String jobid, Props sysProps, Props jobProps, Logger log) {
+    super(jobid, sysProps, jobProps, log);
+    shouldProxy = getSysProps().getBoolean(HadoopSecurityManager.ENABLE_PROXYING, false);
+
+    getJobProps().put(CommonJobProperties.JOB_ID, jobid);
+    shouldProxy = getSysProps().getBoolean(HadoopSecurityManager.ENABLE_PROXYING, false);
+    getJobProps().put(HadoopSecurityManager.ENABLE_PROXYING, Boolean.toString(shouldProxy));
+    obtainTokens = getSysProps().getBoolean(HadoopSecurityManager.OBTAIN_BINARY_TOKEN, false);
+
+    if (shouldProxy) {
+      getLog().info("Initiating hadoop security manager.");
+      try {
+        hadoopSecurityManager = HadoopJobUtils.loadHadoopSecurityManager(getSysProps(), log);
+      } catch (RuntimeException e) {
+        e.printStackTrace();
+        throw new RuntimeException("Failed to get hadoop security manager!" + e.getCause());
+      }
+    }
+
+    tonyConfFile = new File(getWorkingDirectory(), TONY_XML);
+  }
+
+  @Override
+  protected List<String> getClassPaths() {
+    List<String> classPath = super.getClassPaths();
+    classPath.add(tonyConfFile.getParent());
+    return classPath;
+  }
+
+  /**
+   * First refresh the extract
+   * once the status is found, log the results and cancel the job if
+   * the status was an error or a timeout
+   */
+  @Override
+  public void run() throws Exception {
+    getLog().info("Hello world from TensorFlow!");
+    setupHadoopOpts(getJobProps());
+    if (shouldProxy && obtainTokens) {
+      getLog().info("Need to proxy. Getting tokens.");
+      Props props = new Props();
+      props.putAll(getJobProps());
+      props.putAll(getSysProps());
+
+      tokenFile = HadoopJobUtils.getHadoopTokens(hadoopSecurityManager, props, getLog());
+      getJobProps().put("env." + HADOOP_TOKEN_FILE_LOCATION, tokenFile.getAbsolutePath());
+    }
+    super.run();
+  }
+
+  private void setupHadoopOpts(Props props) {
+    if (props.containsKey(HADOOP_GLOBAL_OPTS)) {
+      String hadoopGlobalOps = props.getString(HADOOP_GLOBAL_OPTS);
+      if (props.containsKey(HADOOP_OPTS)) {
+        String hadoopOps = props.getString(HADOOP_OPTS);
+        props.put(HADOOP_OPTS, String.format("%s %s", hadoopOps, hadoopGlobalOps));
+      } else {
+        props.put(HADOOP_OPTS, hadoopGlobalOps);
+      }
+    }
+  }
+
+  @Override
+  public String getJavaClass() {
+    return "com.linkedin.tony.TonyClient";
```

Comment:
Switch to ClusterSubmitter

Reply:
Same as below...ClusterSubmitter is useful for submitting from gateway quickly, but dependency management is much cleaner if hdfs_classpath is configured by users and uses the TonY thin jar.